File: 739-daily-temperatures/daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        n=len(temperatures)
        res=[0]*n
        stack=[]
        for crr_index in range(n):
            while stack!=[] and temperatures[stack[-1]]<temperatures[crr_index]:
                idx=stack.pop()
                diff=crr_index-idx
                res[idx]=diff
            stack.append(crr_index)

        return res

File: 739-daily-temperatures/daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] t) {

        int n=t.length;
        Stack<Integer> stack=new Stack<>();
        int[] res=new int[n];

        for(int i=0;i<n;i++){
            while(stack.isEmpty()==false && t[stack.peek()]<t[i]){
                int idx=stack.pop();
                res[idx]=i-idx;
            }
            stack.push(i);
        }

        return res;
    }
}